docs/src/files.md

```diff
@@ -115,3 +96,47 @@ open(FASTAWriter, my_path) do writer
     end
 end
 ```
+
+However, this latter syntax does not easily extend to different types of IO, such as gzip compressed streams.
+
+### `rdr` and `wtr` macros
+The `rdr` and `wtr` macros use the passed file name to determine the FASTX reader or writer to use - including any compression file extensions.
+Since this both uses heuristics, and the macro is a little opaque to users, it is recommended to use these macros for ephemeral REPL work, and not in packages where the more explicit forms are preferred.
+
+The macro call `rdr"seqs.fna.gz"` expands to
+```julia
+FASTAReader(GzipDecompressorStream(open("seqs.fna.gz"; lock=false)))
+```
+
+To use rdr `rdr` and `wtr` macros with `do`-syntax, use the `defer` function.
```

Comment:
Are we married to `defer`? Might be something useful to bikeshed. I agree we shouldn't use `with` given the new use in `Base`, but `defer` doesn't seem intuitive from a user perspective (though I understand where it's coming from).

Reply:
I'm completely open to suggestions. I like defer because it's short, but it's a little opaque.

Reply:
In my mind, I really want to be able to use existing stuff rather than introducing a new function name. If we do need a new function name, the only thing better I can think of is `apply`, but it's not *much* better.

The primary reason not to just make these iterables so they one can use eg `map` and `foreach` is the need to close the handle, right?

How does `Base.eachline` handle this? It allows you to pass a file name, iterates through lines without needing an explicit `open`, and doesn't need to be closed. But I guess that's only reading, not writing.

To be clear, I don't think any of this should be blocking. `defer` is probably fine, and as-is this is a really big increase in usability. I don't want the perfect to be the enemy of the really good here.

Reply:
> I really want to be able to use existing stuff rather than introducing a new function name
I agree. Not sure what it should be, though.

`Base.eachline` closes the handle when the iterator is exhausted. This is quite elegant, but only works for readers,  not for writers. We could auto-close readers, but I'm a bit wary of guiding users towards not closing their readers, when they really do need to close their writers.